### PR TITLE
Adding a test for the deformation kernels 

### LINKED ISSUE #7499: Adding an example to test the new PolycrystalStoredEnergy kernel
### Description of the enhancement or error report
Adding a test to demonstrate the capability of the PolycrystalStoredEnergy kernel to capture the growth of recrystallized grains at the expense of deformed grains.  

### Rationale for the enhancement or information for reproducing the error
a test cover for the new kernel and demonstration of its capability. 

### Identified impact
None

### PR DESCRIPTION
**Pull request by karim-E-ahmed**
- Test case for the new deformation kernels. 

-closes #7499.

### CODE REVIEW

**Comment by karim86** [comment]
@dschwen I just added a test for the new deformation kernels.

File: modules/phase_field/tests/DeformedGrain/DeformedGrain.i [hunk 1]
```diff
@@ -0,0 +1,120 @@
+[Mesh]
+  type = GeneratedMesh
+  dim = 1
+  nx = 32
```
Comment (dschwen) [inline modules/phase_field/tests/DeformedGrain/DeformedGrain.i]: Do you even need 32 elements here?

File: modules/phase_field/tests/DeformedGrain/DeformedGrain.i [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+  ny = 0
+  nz = 0
+  xmin = 0
+  xmax = 64
+[]
+
+[GlobalParams]
+  block = 0
+  op_num = 2
+  ndef = 1
```
Comment (dschwen) [inline modules/phase_field/tests/DeformedGrain/DeformedGrain.i]: for being used as a global parameter this name is definitely _not_ specific/clear enough

**Comment by karim86** [comment]
I figured it is 1D case anyway, I can trim it down if you like

**Comment by dschwen** [comment]
Or rather increase the grain number

**Comment by karim86** [comment]
@dschwen  I think two grains are enough I'm just showing the growth of a recrystallized grain at the expense of a deformed one. I can add a comment about ndef in the input file or put it under the kernel-material blocks. When I trimmed down the # of elements in half the binary went down only by 10%, I can do it though if you want.

**Comment by dschwen** [comment]
Having more than exactly _one_ grain in each deformed/undeformed set could
help detect an entire new class of bugs. For example, you want to make sure
that you get different values for each deformed grain. You cannot check
that with just one grain.

**Comment by karim86** [comment]
Yes you are absolutely right and I have already done these kinds of tests on my local machine and worked fine. I am here just adding a simple example for test coverage.

**Comment by dschwen** [comment]
We have the tests to ensure that we don't accidentally break our code, so
while it is great you made sure your code works now we want the tests to be
thorough enough to keep it that way.

**Comment by karim86** [comment]
@dschwen Okay, I'll add another test with more grains.

File: modules/phase_field/tests/DeformedGrain/DeformedGrain.i [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+  ny = 0
+  nz = 0
+  xmin = 0
+  xmax = 64
+[]
+
+[GlobalParams]
+  block = 0
+  op_num = 2
+  ndef = 1 # Number of deformed grains.
```
Comment (dschwen) [inline modules/phase_field/tests/DeformedGrain/DeformedGrain.i]: Comment is better than no comment, but I strongly suggest you rename the parameter. Unnecessarily brief names introduce the chance of clashes in the future. I would suggest `deformed_grain_num`

**Comment by karim86** [comment]
@dschwen Not sure why is this example is falling some tests though it did pass others and on my desktop.

**Comment by permcody** [comment]
I've looked into this several times and have theories but no hard proof that my theories are correct. What I suspect is happening is that we are seeing normal floating point errors being magnified by the compiler differences. If a smooth interface falls on an element such that it changes the threshold value of the sampled variable just enough it can have bigger consequences in the grain tracker which treats every element as black or white (either part of a grain or not). Those changes can't be simply ignored because to the grain tracker an element can flip from one integer value to a completely unrelated value on a given element.

@karim86 - If I am correct you have several choices. 
1. Suppress the field output of the GrainTracker Aux Variables so they don't "Exodiff.
2. Create a custom Exodiff comparison file that ignores the GrainTracker Aux Variables.
3. Restrict the test to run on just GCC and/or Clang and avoid testing on Intel.

I'm seeing an exodiff in the "bnds" variable which is more problematic. That one is a little harder to explain away. Finally this test is quite slow (as are most real phase-field tests). You might extend the timeout (I really don't like that), or just skip it in debug mode. I can show you how to do any of these things. Just ask if you need help.

**Comment by dschwen** [comment]
@karim86 please try resizing your simulation cell first! Choose an _odd_ number of elements please. You have an extremely symmetric IC which will have a sharp boundary at _exactly_ the element boundary. I'm 99% sure that this is the problem. We've seen this before.

Try 17*17

**Comment by karim86** [comment]
Okay, I'll try some of your suggestions and see what happens.

**Comment by karim86** [comment]
@permcody I think you are right, so how can I avoid testing on intel and skip the test in debug mode?

**Comment by permcody** [comment]
In your `tests` file:

```
[Tests]
  [./my_test]
    ...
    method = '!DBG'   # Not debug
    compiler = 'GCC CLANG'
  [../]
[]
```

**Comment by karim86** [comment]
I am gonna give it another shot with a different IC and see if it works.

**Comment by karim86** [comment]
Okay, so now it is only a timeout issue. I will just skip that test in debug mode then.

**Comment by karim86** [comment]
@dschwen  this is interesting these are timeout issues with other preexisting tests

**Comment by karim86** [comment]
@dschwen Do you want take a look at this again? I think it is ready to be merged.

**Pull request merged**